doc/source/file_formats.rst

```diff
@@ -1,23 +1,375 @@
+to be added to anchors or changed. Entries consist of an atom name
+followed by an attributes statement. Atoms that are added need to set
+the "PTM_atom" attribute to true and require a valid "element"
+attribute. The "replace" attribute may be set to a (nested) JSON dict,
+listing the atom attributes to be changed and the new values
+corresponding to the modification. Such changes can also be applied to
+atoms already present in the molecular graph, i.e., the 'non-PTM
+atoms'.`
+
+Interactions
+^^^^^^^^^^^^
+
+Interactions are added or changed under the same interaction
+subsections as used for moleculetype. How are interactions removed?
+
+Mappings
+~~~~~~~~
+
+A mapping specifies the conversion from one force field description to
+another. If the transformation is from a higher resolution force field
+to a lower resolution, e.g., from the canonical description to
+Martini, the process is typically called 'forward mapping'. From a
+lower resolution to a higher one is called 'reverse mapping',
+'backward mapping' or 'backmapping'. While forward mapping is
+straightforward, the reverse process requires the addition of details,
+which is typically more difficult, especially if the difference
+between the force field resolutions is larger. The backmapping from
+united atom to all-atom is rather trivial, while the mapping from a
+Cooke lipid model to atomistic is pretty much impossible, and better
+achieved by mapping to and from the intermediate Martini model.
+
+The mapping files use the structure developed for the program backward, given below:
+
+| [ molecule ]
+| [ from ]
+| [ to ]
+| [ martini1 ]
+| [ mapping ]
+| [ atoms ]
+| [ modifiers2 ]
+
+The structure has one top-level section molecule, followed directly by
+the name of the molecule. Thereafter subsections from and to are
+mandatory, specifying the higher-resolution and lower resolution force
+field of the mapping. Next is a subsection bearing the name of the
+lower resolution force field (martini in the example above), with
+contents listing the particles in the order of the corresponding
+moleculetype block. The mapping lists the higher-resolution force
+fields (which is redundant and should be removed). This is followed by
+the actual mapping consisting of an atoms subsection and optional
+geometric modifier subsections assign, out, cis, trans, and chiral.
+
+Atoms
+^^^^^
+
+Only the atoms subsection is used for forward mapping. The contents of
+this section lists the atoms according to the higher resolution force
+field in the same order as specified in the corresponding
+moleculetypes. Each atom consists of a number, the atom name and a
+series of low-resolution particle names. The latter together specify
+the weighing in the mapping, both forward and backward, but are
+easiest interpreted as specifying the position of the higher
+resolution atom by the weighted average of the lower resolution
+particles as given.
+  
+Geometric modifiers
```

Comment:
Do we need this section? Let's leave it in perhaps better to be complete.

Reply:
I'd suggest to remove it, or add a big notice that M2 doesn't use them.